anndata/_io/specs/methods.py

```diff
@@ -301,9 +303,22 @@ def write_list(f, k, elem, dataset_kwargs=MappingProxyType({})):
 @_REGISTRY.register_write(ZarrGroup, np.ndarray, IOSpec("array", "0.2.0"))
 @_REGISTRY.register_write(ZarrGroup, h5py.Dataset, IOSpec("array", "0.2.0"))
 @_REGISTRY.register_write(ZarrGroup, np.ma.MaskedArray, IOSpec("array", "0.2.0"))
+@_REGISTRY.register_write(ZarrGroup, DaskArray, IOSpec("array", "0.2.0"))
+@_REGISTRY.register_write(H5Group, DaskArray, IOSpec("array", "0.2.0"))
 def write_basic(f, k, elem, dataset_kwargs=MappingProxyType({})):
-    """Write methods which underlying library handles nativley."""
-    f.create_dataset(k, data=elem, **dataset_kwargs)
+    """Write methods which underlying library handles natively."""
+    if isinstance(elem, DaskArray):
+        g = f.require_dataset(k, shape=elem.shape, dtype=elem.dtype, **dataset_kwargs)
+        DaskArray.store(elem, g)
```

Comment:
Could this be in a separate function? Like:

```python
@_REGISTRY.register_write(ZarrGroup, DaskArray, IOSpec("array", "0.2.0"))
@_REGISTRY.register_write(H5Group, DaskArray, IOSpec("array", "0.2.0"))
def write_dask_array(f, k, elem, dataset_kwargs=MappingProxyType({})):
     g = f.require_dataset(k, shape=elem.shape, dtype=elem.dtype, **dataset_kwargs)
     DaskArray.store(elem, g)
```

The `write_elem` logic already handles dispatch based on type, so I'd just rely on that rather than putting dispatch flow control within a write method.

Reply:
right! fixed it